### PR TITLE
Add WKB (Well-Known Binary) format

### PR DESCRIPTION
**Pull request by kikuchan**
Implements Well-Known Binary parser and writer.
This also supports EWKB used in PostGIS for example.

### CODE REVIEW

**Comment by jahow** [comment]
Just wanted to say: very clean contribution, thanks!

**Comment by mike-000** [comment]
Just one comment on the documentation.  You document the `hex` option as being used for controlling output, but it has a second purpose through its use in `getType()` which should also be documented, as that will determine whether the loader fetches a url or file as text or ArrayBuffer, for example in

```
new VectorSource({
   format: new WKB(),
   url: 'xxxxxxxx'
})
```

or when used with the `DragAndDrop` interaction.

File: src/ol/format/WKB.js [hunk 1]
```diff
@@ -0,0 +1,866 @@
...
+    super();
+
+    const options = opt_options ? opt_options : {};
+
+    this.splitCollection = Boolean(options.splitCollection);
+
+    this.viewCache_ = null;
+
+    this.hex = options.hex !== false;
+    this.littleEndian = options.littleEndian !== false;
```
Comment (tschaub) [inline src/ol/format/WKB.js:666]: Minor, but if these names are suffixed with `_` (e.g. `littleEndian_`) then they will be available for minification.

File: src/ol/format/WKB.js [hunk 2]
```diff
@@ -0,0 +1,866 @@
...
+      }
+    }
+
+    writer.writeGeometry(
+      transformGeometryWithOptions(geometry, true, options),
+      SRID
+    );
+    const buffer = writer.getBuffer();
+
+    return this.hex !== false ? encodeHexString(buffer) : buffer;
```
Comment (tschaub) [inline src/ol/format/WKB.js:820]:
```suggestion
    return this.hex ? encodeHexString(buffer) : buffer;
```

File: src/ol/format/WKB.js [hunk 3]
```diff
@@ -0,0 +1,866 @@
...
+      ewkb: this.ewkb,
+
+      nodata: {
+        Z: this.nodataZ,
+        M: this.nodataM,
+      },
+    });
+
+    // extract SRID from `dataProjection`
+    let SRID = Number.isInteger(this.SRID) ? Number(this.SRID) : null;
```
Comment (tschaub) [inline src/ol/format/WKB.js:802]: If we want IE support, you could check `typeof this.SRID === 'boolean'` instead.

**Comment by mike-000** [comment]
For consistency with option naming conventions elsewhere in the library `srid` should be lower case in the same way as we use `url` as the option name when specifying a URL.

File: src/ol/format/WKB.js (hunk 3 above)
Comment (mike-000) [reply src/ol/format/WKB.js:802]: `Number.isInteger` is included the the polyfills (`ol/source/IIIF` uses it to check that numbers are integer values).

File: src/ol/format/WKB.js (hunk 1 above)
Comment (kikuchan) [reply src/ol/format/WKB.js]: Thanks! I renamed all of them.

**Comment by kikuchan** [comment]
@mike-000 Thank you. It is really helpful!

**Review by tschaub — APPROVED** [review]

**Comment by tschaub** [comment]
This is such a great contribution.  Thank you, @kikuchan.

**Comment by kikuchan** [comment]
@tschaub Thank you so much!

**Pull request merged**